src/Server/Provision/ServerProvisioner.php

```diff
@@ -234,12 +234,19 @@ private function getIps()
     public function getFilters()
     {
         $choices = $this->config->options();
+        $memory = $this->config->getOption('Memory');
+        $disks = $this->multiChoice($choices, '/Drive Bay ([0-9]+)(.*)/');
+        $addons = $this->addons($this->multiChoice($choices, '/Add On ([0-9]+)/'));
+
+        $preconfiguredMemory = $this->getConfigValues(WhmcsConfig::MEM_BILLING_ID, 'Memory', false, true);
+        $preconfiguredDisks = $this->getConfigValues(WhmcsConfig::DISK_BILLING_IDS, 'Drive Bay', false, false);
+        $preconfiguredAddons = $this->addons($this->getConfigValues(WhmcsConfig::ADDON_BILLING_IDS, 'Add On', false, false));
 
         return [
-            'mem_billing' => $this->config->getOption('Memory'),
+            'mem_billing' => $preconfiguredMemory ?: $memory,
```

Comment:
this is supposed to be the other way around, if there is a configurable option, it should override the preconfigured selections.

Reply:
ah i see, what threw me off was "Our system already has CPU configured this way but we need a way for WHMCS admins to be able to configure all the hardware on a per-product basis without the user being able to override their choice." so it sounded like the user dropdown choice would be moot at that point, but I see that they just won't have the dropdowns if thats the case.

Reply:
That was meant to be "_with_ the user being able to override their choice."